### PR TITLE
U256 DTO fix

### LINKED ISSUE #1082: U256 decoding in TryFrom<&NativeTokenDto> for NativeToken is wrong
## Bug description

`println!("{}", r#"50"#.parse::<U256>().unwrap());`
returns `80`

https://github.com/iotaledger/bee/blob/4710f7891de0d919f2fd39d01394f691eb1f38bf/bee-api/bee-rest-api/src/types/dtos.rs#L894

https://docs.rs/primitive-types/latest/primitive_types/struct.U256.html#method.from_dec_str should be used instead

### PR DESCRIPTION
**Pull request by samuel-rufi**

### CODE REVIEW

File: bee-api/bee-rest-api/src/types/dtos.rs [hunk 1]
```diff
@@ -1484,16 +1480,8 @@ impl TryFrom<&FoundryOutputDto> for FoundryOutput {
...
-                .circulating_supply
-                .0
-                .parse::<U256>()
-                .map_err(|_| Error::InvalidField("circulating_supply"))?,
-            value
-                .maximum_supply
-                .0
-                .parse::<U256>()
-                .map_err(|_| Error::InvalidField("maximum_supply"))?,
+            U256::from_dec_str(value.circulating_supply.0).map_err(|_| Error::InvalidField("circulating_supply"))?,
```
Comment (thibault-martinez) [inline bee-api/bee-rest-api/src/types/dtos.rs]: I would also assume that U256::to_string uses hex

Comment (samuel-rufi) [reply bee-api/bee-rest-api/src/types/dtos.rs]: Let me check the code

**Review by Thoralf-M — APPROVED** [review]

File: bee-api/bee-rest-api/src/types/dtos.rs (hunk 1 above)
Comment (Thoralf-M) [reply bee-api/bee-rest-api/src/types/dtos.rs]: Seems to be fine as it's now
```Rust
let native_tokens = NativeTokenDto::try_from(&NativeToken::new(TokenId::new(token_id), U256::from(50))?).unwrap();
println!("{}", U256::from(50));
println!("{:?}", native_tokens);
println!("{}", serde_json::to_string(&native_tokens)?);
```
prints
```bash
50
NativeTokenDto { token_id: TokenIdDto("08e68f7616cd4948efebc6a77c4f93aed770ac53860100000000000000000000000000000000"), amount: U256Dto("50") }
{"id":"08e68f7616cd4948efebc6a77c4f93aed770ac53860100000000000000000000000000000000","amount":"50"}
```

Comment (thibault-martinez) [reply bee-api/bee-rest-api/src/types/dtos.rs]: Doesn't make any sense :trollface:
Finishing a PR, I'll look in a bit.

Comment (Thoralf-M) [reply bee-api/bee-rest-api/src/types/dtos.rs]: For serialization it also uses hex, but in the DTO it's only a String and there we use `.to_string()`
https://github.com/iotaledger/bee/blob/aacf78d48d01d36cadacb2f4b8acc7e9c1922ae3/bee-api/bee-rest-api/src/types/dtos.rs#L880

Comment (thibault-martinez) [reply bee-api/bee-rest-api/src/types/dtos.rs]: Ah alright, spoke too quickly, my bad thanks 👍

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**